app-prefix/README.md

```diff
@@ -0,0 +1,46 @@
+---
+creation_date: 2024-09-26
+issues:
+- https://github.com/giantswarm/workload-clusters-fleet/pull/802
+last_review_date: 2024-09-26
+owners:
+- https://github.com/orgs/giantswarm/teams/team-honeybadger
+state: review
+summary: Decide whether the apps in CAPI cluster should have or not a prefix, and if we need to enforce it.
+---
+
+# App prefix
+
+## Intro
+
+When we moved to Cluster API, we decided that all resources would live the organization namespace. That decision means apps, including the cluster app, from different clusters coexist in the same namespace. It also means, it forces us to name the apps differently to avoid conflicts.
+
+## Scope
+
+This affects `kubectl gs` since we have a mix of functionality. In `gitops` subcommand we append the cluster id as a prefix on the output but in `template` subcommand the user needs too add explicitly the `--app-name` with the prefix on it.
+
+When users use the GitOps template, we advocate relying on the `namePrefix` variable to prefix the apps and config maps automatically.
+
+Lately, happa has also prefixed the apps automatically.
+
+We don't validate or mutate the app name via the management cluster API.
+
+## Solution
```

Comment:
Regarding my comment before and independence of apps and clusters. An alterntive solution would be to do something like pod names for replicasets. Just append some unique identifier to the name of the app.

Reply:
But then it is same problem the app is not the same as you defined in the manifest. Or we need to create a new resource for the `app instance` entity